python_modules/dagster/dagster/core/definitions/time_window_partitions.py

```diff
@@ -141,17 +148,14 @@ def time_window_for_partition_key(self, partition_key: str) -> TimeWindow:
             execution_timezone=self.timezone,
         )
 
-        return TimeWindow(next(iterator), next(iterator))
-
-    def start_time_for_partition_key(self, partition_key: str) -> datetime:
-        return pendulum.instance(datetime.strptime(partition_key, self.fmt), tz=self.timezone)
-
-    def get_default_partition_mapping(self):
-        from dagster.core.definitions.time_window_partition_mapping import (
-            TimeWindowPartitionMapping,
-        )
+        prev_time = next(iterator)
+        while prev_time.timestamp() < start_timestamp:
```

Comment:
don't fully understand why this is necessary (would have expected the first value of iterator to always be start_timestamp), but I accept that time iterator functions are complicated and that there's probably a good reason for this.